de-lib/src/main/java/org/iplantc/de/analysis/client/presenter/AnalysesPresenterImpl.java

```diff
@@ -448,7 +475,83 @@ public void onSuccess(AnalysisStepsInfo result) {
 
     }
 
+
     AnalysisStepsInfoDialog getAnalysisStepsDialog() {
         return new AnalysisStepsInfoDialog(analysisStepView);
     }
+
+    private void shareWithSupport(Analysis selectedAnalysis, final Splittable parent) {
+        AnalysisPermission ap = shareFactory.analysisPermission().as();
+        ap.setId(selectedAnalysis.getId());
+        ap.setPermission(PermissionValue.read.toString());
+        AnalysisSharingRequest asr = shareFactory.AnalysisSharingRequest().as();
+        asr.setUser(deProperties.getSupportUser());
+        asr.setAnalysisPermissions(Arrays.asList(ap));
+        AnalysisSharingRequestList listRequest = shareFactory.AnalysisSharingRequestList().as();
+        listRequest.setAnalysisSharingRequestList(Arrays.asList(asr));
+        analysisService.shareAnalyses(listRequest, new AnalysisCallback<String>() {
+            @Override
+            public void onFailure(Integer statusCode, Throwable exception) {
+                ErrorHandler.post(exception);
+            }
+
+            @Override
+            public void onSuccess(String result) {
+                emailSupport(parent);
+            }
+        });
+    }
+
+    @Override
+    public void onUserSupportRequested(AnalysisUserSupportRequestedEvent event) {
+        final Analysis value = event.getValue();
+        final AnalysisUserSupportDialog ausd = new AnalysisUserSupportDialog(value);
```

Comment:
Just because I've been looking at code splits...

It'd be nice if this class could have a `@Inject AsyncProviderWrapper<AnalysisUserSupportDialog>` and then use that provider to get a dialog and call something to pass in the value then. That way, the code for the dialog doesn't go into the initial fragment.